src/js/components/search-field/SearchField.jsx

```diff
@@ -1,16 +1,15 @@
 import React, { Component } from 'react';
 import { render } from 'react-dom';
+import { connect } from 'react-redux';
+import { searchPhraseChange } from '../../actions/actionCreator';
+import { bindActionCreators } from 'redux';
 import { FormGroup, ControlLabel, FormControl } from 'react-bootstrap';
 import './searchfield.style.less';
 
 class SearchField extends Component {
 
-  state = {
-    value: ''
-  };
-
   handleChange = (event) => {
-    this.setState({ value: event.target.value });
+    this.props.searchPhraseChange(event.target.value);
```

Comment:
Here it works great as you're working on static data. But let's think if you'd have to perform an API call to get those data into your state then you'd hit API at every keypress.

So just a sugesstion maybe if you want try to debounce this event and trigger this after user stops typing (delay 400-700 ms)

a small helper: https://www.youtube.com/watch?v=KXao_qwl05k